src/GSoft.Extensions.MediatR.Analyzers/NamingConventionAnalyzer.cs

```diff
@@ -0,0 +1,277 @@
+﻿using System.Collections.Immutable;
+using GSoft.Extensions.MediatR.Analyzers.Internals;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Diagnostics;
+
+namespace GSoft.Extensions.MediatR.Analyzers;
+
+[DiagnosticAnalyzer(LanguageNames.CSharp)]
+public sealed class NamingConventionAnalyzer : DiagnosticAnalyzer
+{
+    internal static readonly DiagnosticDescriptor UseCommandOrQuerySuffixRule = new DiagnosticDescriptor(
+        id: RuleIdentifiers.UseCommandOrQuerySuffix,
+        title: "Name should end with Command or Query",
+        messageFormat: "Name should end with Command or Query",
+        category: RuleCategories.Naming,
+        defaultSeverity: DiagnosticSeverity.Warning,
+        isEnabledByDefault: true,
+        description: null,
+        helpLinkUri: RuleIdentifiers.HelpUri);
+
+    internal static readonly DiagnosticDescriptor UseCommandHandlerOrQueryHandlerSuffixRule = new DiagnosticDescriptor(
+        id: RuleIdentifiers.UseCommandHandlerOrQueryHandlerSuffix,
+        title: "Name should end with CommandHandler or QueryHandler",
+        messageFormat: "Name should end with CommandHandler or QueryHandler",
+        category: RuleCategories.Naming,
+        defaultSeverity: DiagnosticSeverity.Warning,
+        isEnabledByDefault: true,
+        description: null,
```

Comment:
Is it necessary to specify `null` here? Can this parameter simply be omitted or not?

Reply:
Not necessary, description is `null` by default. I removed all of them